glusterapi/volume.py

```diff
@@ -1,132 +1,263 @@
-from common import BaseAPI
-from exceptions import GlusterApiInvalidInputs
+"""This module contains the python  glusterd2 volume api's implementation."""
+import httplib
+import json
 
+from glusterapi.common import BaseAPI, validate_uuid, validate_volume_name
+from glusterapi.exceptions import GlusterApiInvalidInputs
 
-class TransportType:
+
+class TransportType(object):
     TCP = "tcp"
     RDMA = "rdma"
     Both = "tcp,rdma"
 
+    def check(self, transport_type):
+        if transport_type not in (self.TCP, self.RDMA, self.Both):
+            raise GlusterApiInvalidInputs(
+                "Transport type %s not supported" % transport_type)
+
+
+def validate_brick(bricks=None):
+    """
+    Validate brick pattern.
+
+    :param bricks: (string) in the form of nodeid:brickpath
+    :return brick_req: (list) list of bricks
+    """
+    brick_req = []
+    if bricks is None:
+        return None
+    for brick in bricks:
+        brk = brick.split(":")
+        if len(brk) != 2:
+            return None
+        if validate_uuid(brk[0]) is False:
+            return None
+        req = dict()
+        req['peerid'] = brk[0]
+        req['path'] = brk[1]
+        brick_req.append(req)
+    return brick_req
+
 
 class VolumeApis(BaseAPI):
-    def volume_create(self, volname, size=0, bricks=[],
-                      transport=TransportType.TCP,
-                      replica=None,
-                      disperse=None,
-                      arbiter=None,
-                      force=False):
+    def volume_create(self, volume_name="", bricks=None,
```

Comment:
Can we make volume_name mandatory? Because this will anyway fail eventually when you post the data. I'm sorry I missed mentioning this earlier.

Reply:
@aravindavk when we can expect https://github.com/gluster/glusterd2/pull/876 to be merged?

Reply:
Ack! Didn't know about that.